azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub.py

```diff
@@ -45,3 +46,42 @@ def _run_op(self, op):
             )
         else:
             self.continue_op(op)
+
+
+class HandleTwinOperations(PipelineStage):
+    """
+    PipelineStage which handles twin operations. In particular, it converts twin GET and PATCH
+    operations into SendIotRequestAndWaitForResponse operations.  This is done at the IotHub level because
+    there is nothing transport-specific about this code.  The transport-specific implementation
```

Comment:
> transport-specific [](start = 20, length = 19)

Even if we end up keeping "transport" around in the API for legacy purposes, I don't think we should keep referring to it internally. Makes no sense to anyone who doesn't know the history of the SDK. We're talking about protocols here.

Reply:
I'm opening a thread.  protocol is a very generic word.  transport has a very specific definition that doesn't quite match either.  The last T in MQTT stands for Transport, but the last P in HTTP and AMQP stands for Protocol.  But, in the OSI 7-layer stack, HTTP is an Application Protocol and AMQP and MQTT would be Session Protocols.  In the case of MQTT, there are at least 4 protocols involved - MQTT, TCP, IP, and Ethernet (or whatever physical connection you're using).  Saying "the protocol" is meaningless unless we have agreement on what "a protocol" is.  

---
In reply to: [288349774](https://github.com/Azure/azure-iot-sdk-python-preview/pull/96#discussion_r288349774) [](ancestors = 288349774)